utils/key.go

```diff
@@ -0,0 +1,32 @@
+// (c) 2021-2022, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package utils
+
+import (
+	"crypto/ecdsa"
+	"io"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+)
+
+type Key struct {
+	Address    common.Address
+	PrivateKey *ecdsa.PrivateKey
+}
+
+func NewKey(rand io.Reader) (*Key, error) {
```

Comment:
Why accept the reader if it's always `crypto/rand.Reader` (at least in the code I can see)? I understand wanting stable keys in tests, but it doesn't look like that's being done and this risks someone may accidentally pass a non-secure source.

Reply:
I think this is fine since it's the same in the existing code it was adapted from and also in ecdsa.GenerateKey
Additionally it's recommended to receive interfaces in golang as opposed to concrete types

Reply:
> receive interfaces in golang as opposed to concrete types

I mean not receive anything at all and just use `crypto/rand.Reader` inside the function. I'm happy with whatever you decide here—consider my original question as just a prompt to spur thinking.